pkg/network/tracer/tracer.go

```diff
@@ -850,3 +851,17 @@ func newUSMMonitor(c *config.Config, tracer connection.Tracer) *usm.Monitor {
 
 	return monitor
 }
+
+// GetNetworkID retrieves the vpc_id (network_id) from IMDS
+func (t *Tracer) GetNetworkID() (string, error) {
+	id := ""
+	err := kernel.WithRootNS(kernel.ProcFSRoot(), func() error {
+		var err error
+		id, err = ec2.GetNetworkID(context.TODO())
+		return err
+	})
+	if err != nil {
+		return "", err
+	}
+	return id, nil
```

Comment:
We should cache the id here to avoid another call to the metadata endpoint.

Reply:
I believe either way this will only get called once, since this is hit via the `Init()` method of the PA check which happens only at startup

Reply:
although if multiple checks are calling this it will indeed be called multiple times, so we can save 2 RPCs by caching this

Reply:
upon further reflection I think I would rather not cache this, because:
- we are maintaining the same call pattern as before
- we are only calling max 3 times on startup (if process/container/network checks are all enabled)
- anecdotally I have seen a failure to fetch the IMDSv2 token sometimes when testing. I am going to follow up with ASC about this, but I am wary to not populate this value if the first request fails